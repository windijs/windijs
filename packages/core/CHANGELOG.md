# @windijs/core

## 1.1.0-beta.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

### Patch Changes

- Updated dependencies [[`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33)]:
  - @windijs/helpers@1.1.0-beta.0

## 1.0.5

## 1.0.4

### Patch Changes

- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3
  - @windijs/helpers@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/helpers@1.0.2
  - @windijs/shared@1.0.2

## 1.0.2

- chore: release v1.0.2

## 1.0.1

### Patch Changes

- chore: release v1.0.1
- Updated dependencies
  - @windijs/helpers@1.0.1
  - @windijs/shared@1.0.1

## 1.0.0

### Patch Changes

- @windijs/helpers@1.0.0
