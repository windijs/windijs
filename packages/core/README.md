# @windi/core

Core apis of windijs.

## Basic

- use
- meld
- guard
- variant
- media
- bind
- useVariant
- useMedia
- createUtility
- createVariant
- createMedia
- createScreenVariants
- createDarkModeVariants
- createOrientationVariants
- createMotionVariants
- handleConfig

## Handlers

- cssHandler
- callHandler
- colorHandler
- configHandler
- genericHandler
- numberHandler
- pxHandler
- remHandler
- degHandler
- msHandler
- spacingHandler
- fractionHandler

## Builders

- buildProperty
- buildStatic
- buildColor
- buildContainer
- buildNotHidden
- buildFontSize
- buildFlexDirection
- buildFlexStretch
- buildReverse
- buildDivideY
- buildDivideX
- buildSpaceBetweenY
- buildSpaceBetweenX
- buildFilter
- buildBackdropFilter
- buildTransform
- buildTransition
- buildKeyframes
- buildGradientDirection
- buildLinearGradient
- buildGradientFrom
- buildGradientVia
- buildGradientTo
- buildDivideColor
- buildDivideOpacity
- buildDivideStyle
- buildRingWidth
- buildBoxShadowSize
- buildBoxShadowColor
- buildImageRendering
- buildPlaceholder
- buildWritingMode

## Prefixers

- prefixer
- prefixKeyframes
- prefixAnimation
- prefixImageRendering
- prefixNotHidden
- prefixPlaceholder
- prefixWritingMode