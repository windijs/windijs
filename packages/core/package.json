{
  "name": "@windijs/core",
  "version": "1.1.0-beta.2",
  "description": "@windijs/core",
  "main": "dist/core.js",
  "module": "dist/core.mjs",
  "types": "dist/core.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    ".": {
      "import": "./dist/core.mjs",
      "require": "./dist/core.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "dependencies": {
    "@windijs/shared": "1.0.4-beta.0",
    "@windijs/helpers": "1.1.0-beta.2"
  },
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/core"
  }
}
