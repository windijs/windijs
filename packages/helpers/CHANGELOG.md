# @windijs/helpers

## 1.1.2

### Patch Changes

- [`39b28846`](https://github.com/windijs/windijs/commit/39b28846c780dc1c39ca30abfcd8e9c554aadb63) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: move dts generators to plugin-utils

## 1.1.1

### Patch Changes

- [`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: export package.json

* [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2) Thanks [@voorjaar](https://github.com/voorjaar)! - build: switch to es2018 for better tree shaking

* Updated dependencies [[`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3), [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2)]:
  - @windijs/shared@1.0.5

## 1.1.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

### Patch Changes

- [`8ab80baf`](https://github.com/windijs/windijs/commit/8ab80bafda9ab832d8d53f287e200f7fd497b7b6) Thanks [@voorjaar](https://github.com/voorjaar)! - chore: new versions

* [`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: remove circular devDependencies

- [`e9a948cd`](https://github.com/windijs/windijs/commit/e9a948cd44d003a2c1e91ea2927658f59e3746dc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new resetStyleMeta helper

* [`9940db66`](https://github.com/windijs/windijs/commit/9940db664965f9e7c04b8e6831cb035b79f2b212) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesDts with theme

- [`d0d14195`](https://github.com/windijs/windijs/commit/d0d14195bea0f846097a9738da82278ca97468bc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new handler types

* [`6a71ee76`](https://github.com/windijs/windijs/commit/6a71ee76737a9309b31b06adc33a695461d1583d) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: add dtsSetup and dtsHandler

- [`da5e3466`](https://github.com/windijs/windijs/commit/da5e3466071e7c780b3875e177dad87e14495d3c) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesJs with theme

* [`ba98151d`](https://github.com/windijs/windijs/commit/ba98151d7050fe6832999cf4c7ce0bec48d26fe0) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: nested StyleObject in ArrayHelper

* Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4

## 1.1.0-beta.5

### Patch Changes

- [`da5e3466`](https://github.com/windijs/windijs/commit/da5e3466071e7c780b3875e177dad87e14495d3c) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesJs with theme

* [`ba98151d`](https://github.com/windijs/windijs/commit/ba98151d7050fe6832999cf4c7ce0bec48d26fe0) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: nested StyleObject in ArrayHelper

## 1.1.0-beta.4

### Patch Changes

- [`9940db66`](https://github.com/windijs/windijs/commit/9940db664965f9e7c04b8e6831cb035b79f2b212) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: genUtilitiesDts with theme

## 1.1.0-beta.3

### Patch Changes

- [`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: remove circular devDependencies

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/shared@1.0.4-beta.1

## 1.1.0-beta.2

### Patch Changes

- [`d0d14195`](https://github.com/windijs/windijs/commit/d0d14195bea0f846097a9738da82278ca97468bc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new handler types

* [`6a71ee76`](https://github.com/windijs/windijs/commit/6a71ee76737a9309b31b06adc33a695461d1583d) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: add dtsSetup and dtsHandler

* Updated dependencies [[`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4-beta.0

## 1.1.0-beta.1

### Patch Changes

- [`8ab80baf`](https://github.com/windijs/windijs/commit/8ab80bafda9ab832d8d53f287e200f7fd497b7b6) Thanks [@voorjaar](https://github.com/voorjaar)! - chore: new versions

* [`e9a948cd`](https://github.com/windijs/windijs/commit/e9a948cd44d003a2c1e91ea2927658f59e3746dc) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: new resetStyleMeta helper

## 1.1.0-beta.0

### Minor Changes

- [`a8e51fe2`](https://github.com/windijs/windijs/commit/a8e51fe2d193dc7e6b450fb4e7b5ad6ab330bd33) Thanks [@voorjaar](https://github.com/voorjaar)! - refactor: new handler api with meta data

## 1.0.3

### Patch Changes

- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3

## 1.0.2

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/shared@1.0.2

## 1.0.1

### Patch Changes

- chore: release v1.0.1
- Updated dependencies
  - @windijs/shared@1.0.1

## 1.0.0
