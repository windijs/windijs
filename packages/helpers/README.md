# @windi/helpers

Operations related to css.

## CSS In JS

- $
- css
- prop
- atomic
- unify
- inline
- bundle
- injectCSS
- escapeCSS
- applyVariant
- cssInJsLoader
- useArrayHelper
- useStyleLoader

## Namer

- alphaNamer
- atomicNamer
- hashNamer
- useNamer
- nameStyle

## Meta

- getMeta
- getUid
- resetMeta
- pushMetaProp
- updateMetaType

## Utils

- isStyleObject
- getStyleVariants
- getStyleProps
- getStyleIdent
- baseStyleTarget
- baseStyleHandler
- getFirstVar
- useProxy
- setProxy
- isProxy

## Color Process

- digitToHEX
- rgbToHEX
- hexToRGB
- sliceColor
- rgbToHSL
- hueToRGB
- hslToRGB
- hwbToRGB
- adjustHue
- adjustSaturation
- adjustLightness
- adjustRed
- adjustGreen
- adjustBlue
- adjustAlpha
- mixColor
- subMixColor
- colorLuminance
- getLightColor
- getDarkColor
- Color
    - static hex
    - static rgb
    - static rgba
    - static hsl
    - static hsla
    - static hwb
    - static mix
    - static subcolormix
    - get hex
    - get rgb
    - get rgba
    - get hsl
    - get hsla
    - get hwb
    - get red
    - get blue
    - get green
    - get hue
    - get saturation
    - get lightness
    - get alpha
    - get opacity
    - get whiteness
    - get blacknness
    - get ieHexStr
    - invert()
    - adjustRed()
    - adjustGreen()
    - adjustBlue()
    - adjustHue()
    - adjustSaturation()
    - adjustLightness()
    - adjustAlpha()
    - complement()
    - saturate()
    - desaturate()
    - grayscale()
    - lighten()
    - darken()
    - opacify()
    - transparentize()
    - fadeIn()
    - fadeOut()
    - adjust()
    - scale()
    - change()
    - lightenSet()
    - darkenSet()
    - desaturateSet()
    - complementSet()
    - invertSet()

## CSS Units

- color
- percent
- deg, grad, rad, turn
- s, ms
- fr
- $in
- dpi, dpcm, dppx, x
- px, pc, pt, cm, mm, Q, ch, ex, em, rem, vw, vh, vmax, vmin

## CSS Functions

- filters
- transforms
- quote
- attr
- url
- $var
- path
- hwb
- dropShadow
- counters
- circle
- ellipse
- inset
- polygon
- matrix
- matrix3d
- perspective
- rotate
- rotate3d
- rotateX
- rotateY
- rotateZ
- scale
- scale3d
- scaleX
- scaleY
- scaleZ
- skew
- skewX
- skewY
- translate
- translate3d
- translateX
- translateY
- translateZ
- steps
- calc
- clamp
- max
- min
- abs
- sign
- blur
- brightness
- contrast
- grayscale
- invert
- opacity
- saturate
- sepia
- rgb
- rgba
- hsl
- hsla
- counter
- env
- minmax
- repeat
- hueRotate
- fitContent
- cubicBezier
- linearGradient
- radialGradient
- conicGradient
- repeatingConicGradient
- repeatingLinearGradient
- repeatingRadialGradient

## CSS Math

- add
- sub
- mul
- div
- prec