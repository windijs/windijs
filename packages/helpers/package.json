{
  "name": "@windijs/helpers",
  "version": "1.1.2",
  "description": "@windijs/helpers",
  "main": "dist/helpers.js",
  "module": "dist/helpers.mjs",
  "types": "dist/helpers.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/helpers.mjs",
      "require": "./dist/helpers.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "dependencies": {
    "@windijs/config": "1.0.7",
    "@windijs/shared": "1.0.5"
  },
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/helpers"
  }
}
