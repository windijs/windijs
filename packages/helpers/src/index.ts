// TODO: maybe add more helpers reference sass?
// https://sass-lang.com/documentation/modules

export * from "./types";
export * from "./build";
export * from "./config";
export * from "./color";
export * from "./common";
export * from "./css";
export * from "./data";
export * from "./dollar";
export * from "./escape";
export * from "./funcs";
export * from "./math";
export * from "./meta";
export * from "./namer";
export * from "./proxy";
export * from "./unit";
