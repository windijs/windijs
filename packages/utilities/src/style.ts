import { createUtility, stylePropertyHandler } from "@windi/core";

export const style = createUtility("style").use(stylePropertyHandler()).init();
