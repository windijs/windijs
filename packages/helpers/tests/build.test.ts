import { variant } from "@windijs/core";
import { atomic, atomicNamer, buildCSS, bundle, hashNamer, unify, useNamer } from "@windijs/helpers";
import { style } from "@windijs/style";
import { bg, border, rounded, text } from "@windijs/utilities";
import { dark, hover, sm } from "@windijs/variants";

test("unify with utilities", () => {
  expect(unify(".test", bg.blue[500], text.lg, sm(border, rounded, bg.blue[400]))).toMatchSnapshot();

  expect(unify(".test2", bg.blue[500], text.lg, [rounded.md, border[2]], sm(border, rounded, bg.blue[400]))).toMatchSnapshot();
});

test("unify with utility object", () => {
  expect(
    unify({
      ".test": [bg.blue[500], text.lg],
      ".button": [bg.rose[500], rounded, border],
    })
  ).toMatchSnapshot();

  expect(
    unify(
      {
        ".test": [bg.blue[500], text.lg],
        ".button": [bg.rose[500], rounded, border],
      },
      {
        p: [text.lg],
      }
    )
  ).toMatchSnapshot("a");

  expect(
    unify(
      {
        ".test": [bg.blue[500], text.lg],
        ".button": [bg.rose[500], rounded, border],
      },
      {
        p: text.lg,
      }
    )
  ).toMatchSnapshot("a");
});

test("atomic", () => {
  expect(atomic(bg.blue[500], text.lg, sm(border, rounded, bg.blue[300]))).toMatchSnapshot();
});

test("atomic with nested variant", () => {
  expect(
    atomic(bg.blue[500], text.lg, hover(bg.yellow[300]), sm(border, rounded, bg.blue[300], hover(bg.blue[600]), dark(hover(bg.blue[800]))))
  ).toMatchSnapshot();
});

test("atomic with atomicNamer", () => {
  useNamer(atomicNamer);

  expect(atomic(bg.blue[500], text.lg, sm(border, rounded, bg.blue[300]))).toMatchSnapshot();
});

test("atomic with nested variant and atomicNamer", () => {
  useNamer(atomicNamer);

  expect(
    atomic(bg.blue[500], text.lg, hover(bg.yellow[300]), sm(border, rounded, bg.blue[300], hover(bg.blue[600]), dark(hover(bg.blue[800]))))
  ).toMatchSnapshot();
});

test("atomic with hashNamer", () => {
  useNamer(hashNamer);

  expect(atomic(bg.blue[500], text.lg, sm(border, rounded, bg.blue[300]))).toMatchSnapshot();
});

test("atomic with empty variant", () => {
  expect(atomic(sm())).toEqual("");
  expect(atomic()).toEqual("");
});

test("bundle with nested media", () => {
  const bundled = bundle(
    hover(variant("@media (hover: hover)", style.border["2px solid black"], variant("@media (color)", style.borderColor["#036"])))
  );
  expect(bundled).toMatchSnapshot();
});

test("build keyframes", () => {
  expect(
    buildCSS({
      "@-webkit-keyframes": {
        from: { opacity: 0 },
        to: { opacity: 1 },
      },
      "@keyframes slideRight": {
        from: { opacity: 0 },
        to: { opacity: 1 },
      },
      ".container": {
        animationName: "slideRight",
      },
    })
  ).toMatchSnapshot();
});

test("build same css property", () => {
  expect(
    buildCSS({
      ".flex": {
        display: ["flex", "-ms-flex"],
      },
    })
  ).toMatchSnapshot();
});

test("build nested css", () => {
  expect(
    buildCSS({
      ".flex": {
        ".b": {
          fontSize: "large",
        },
        "&": {
          display: ["flex", "-ms-flex"],
        },
        "& .c": {
          fontSize: "1rem",
        },
        "@media print": {
          "@font-face": {
            fontFamily: "monospace",
          },
        },
        "@media (min-width: 900px)": {
          ".abc": {
            fontSize: "2rem",
          },
          "&.def": {
            fontSize: "3rem",
          },
        },
        "@supports (display: flex)": {
          "&": {
            fontSize: "4rem",
          },
          "@media (min-width: 800px)": {
            body: {
              background: "red",
            },
            "& > p": {
              padding: "1rem",
            },
          },
        },
      },
    })
  ).toMatchSnapshot();
});

test("build font-face", () => {
  expect(
    buildCSS({
      "@font-face": {
        fontFamily: "monospace",
        src: "url(webfont.eot)",
      },
    })
  ).toMatchSnapshot();

  expect(
    buildCSS({
      "@font-face": [
        {
          fontFamily: "MyWebFont",
          src: "url(webfont.eot)",
        },
        {
          fontFamily: "MySecondFont",
          src: "url(webfont2.eot)",
        },
      ],
    })
  ).toMatchSnapshot();
});

test("build media queries", () => {
  expect(
    buildCSS({
      "@media screen and (min-width: 900px)": {
        article: {
          padding: "1rem 3rem",
        },
      },
      "@supports (display: flex)": {
        "@media screen and (min-width: 900px)": {
          article: {
            display: "flex",
          },
        },
      },
      "@media print": {
        "@font-face": {
          fontFamily: "MySecondFont",
          src: "url(webfont2.eot)",
        },
      },
      "@media (prefers-color-scheme: dark)": {
        "@media (min-width: 900px)": {
          article: {
            display: "block",
          },
        },
      },
    })
  ).toMatchSnapshot();
});

test("should throw errors", () => {
  expect(() =>
    buildCSS({
      padding: "1rem",
    })
  ).toThrowError();

  expect(() =>
    buildCSS({
      "@media print": {
        padding: "1rem",
      },
    })
  ).toThrowError();

  expect(() =>
    buildCSS({
      "@font-face": {
        ".nested": {
          fontSize: "large",
        },
      },
    })
  ).toThrowError();
});

test("build global media rules", () => {
  expect(
    buildCSS({
      "@namespace": "url(XML-namespace-URL)",
      "@charset": '"utf-8"',
      "@import": "url(http://mysite.com/custom.css)",
      "@layer": "utilities",
      "@layer utilities": {
        ".padding-sm": {
          padding: "0.5rem",
        },
        ".padding-lg": {
          padding: "0.8rem",
        },
      },
    })
  ).toMatchSnapshot();
});
