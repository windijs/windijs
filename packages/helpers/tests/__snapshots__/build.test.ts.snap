// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atomic 1`] = `
".a {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.b {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

@media (min-width: 640px) {
  .c {
    border-width: 1px;
  }
  .d {
    border-radius: 0.25rem;
  }
  .e {
    background-color: rgba(147, 197, 253, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
}"
`;

exports[`atomic with atomicNamer 1`] = `
".bg\\\\.blue\\\\.500 {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.text\\\\.lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

@media (min-width: 640px) {
  .border {
    border-width: 1px;
  }
  .rounded {
    border-radius: 0.25rem;
  }
  .bg\\\\.blue\\\\.300 {
    background-color: rgba(147, 197, 253, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
}"
`;

exports[`atomic with hashNamer 1`] = `
".w-ixsi2d {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.w-fhiqyx {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

@media (min-width: 640px) {
  .w-4mwz7n {
    border-width: 1px;
  }
  .w-3cecw {
    border-radius: 0.25rem;
  }
  .w-1ho46uf {
    background-color: rgba(147, 197, 253, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
}"
`;

exports[`atomic with nested variant 1`] = `
".a {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.b {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.f:hover {
  background-color: rgba(253, 224, 71, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

@media (min-width: 640px) {
  .c {
    border-width: 1px;
  }
  .d {
    border-radius: 0.25rem;
  }
  .e {
    background-color: rgba(147, 197, 253, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
  .g:hover {
    background-color: rgba(37, 99, 235, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
  @media (prefers-color-scheme: dark) {
    .h:hover {
      background-color: rgba(30, 64, 175, var(--w-bg-opacity));
      --w-bg-opacity: 1;
    }
  }
}"
`;

exports[`atomic with nested variant and atomicNamer 1`] = `
".bg\\\\.blue\\\\.500 {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.text\\\\.lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.bg\\\\.yellow\\\\.300:hover {
  background-color: rgba(253, 224, 71, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

@media (min-width: 640px) {
  .border {
    border-width: 1px;
  }
  .rounded {
    border-radius: 0.25rem;
  }
  .bg\\\\.blue\\\\.300 {
    background-color: rgba(147, 197, 253, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
  .bg\\\\.blue\\\\.600:hover {
    background-color: rgba(37, 99, 235, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
  @media (prefers-color-scheme: dark) {
    .bg\\\\.blue\\\\.800:hover {
      background-color: rgba(30, 64, 175, var(--w-bg-opacity));
      --w-bg-opacity: 1;
    }
  }
}"
`;

exports[`build font-face 1`] = `
"@font-face {
  font-family: monospace;
  src: url(webfont.eot);
}"
`;

exports[`build font-face 2`] = `
"@font-face {
  font-family: MyWebFont;
  src: url(webfont.eot);
}

@font-face {
  font-family: MySecondFont;
  src: url(webfont2.eot);
}"
`;

exports[`build global media rules 1`] = `
"@namespace url(XML-namespace-URL);

@charset \\"utf-8\\";

@import url(http://mysite.com/custom.css);

@layer utilities;

@layer utilities {
  .padding-sm {
    padding: 0.5rem;
  }
  .padding-lg {
    padding: 0.8rem;
  }
}"
`;

exports[`build keyframes 1`] = `
"@-webkit-keyframes {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes slideRight {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.container {
  animation-name: slideRight;
}"
`;

exports[`build media queries 1`] = `
"@media screen and (min-width: 900px) {
  article {
    padding: 1rem 3rem;
  }
}

@supports (display: flex) {
  @media screen and (min-width: 900px) {
    article {
      display: flex;
    }
  }
}

@media print {
  @font-face {
    font-family: MySecondFont;
    src: url(webfont2.eot);
  }
}

@media (prefers-color-scheme: dark) {
  @media (min-width: 900px) {
    article {
      display: block;
    }
  }
}"
`;

exports[`build nested css 1`] = `
".flex .b {
  font-size: large;
}

.flex {
  display: flex;
  display: -ms-flex;
}

.flex .c {
  font-size: 1rem;
}

@media print {
  @font-face {
    font-family: monospace;
  }
}

@media (min-width: 900px) {
  .flex .abc {
    font-size: 2rem;
  }
  .flex.def {
    font-size: 3rem;
  }
}

@supports (display: flex) {
  .flex {
    font-size: 4rem;
  }
  @media (min-width: 800px) {
    .flex body {
      background: red;
    }
    .flex > p {
      padding: 1rem;
    }
  }
}"
`;

exports[`build same css property 1`] = `
".flex {
  display: flex;
  display: -ms-flex;
}"
`;

exports[`bundle with nested media 1`] = `
Map {
  "&:hover" => Object {
    "@media (hover: hover)": Object {
      "@media (color)": Object {
        "borderColor": "#036",
      },
      "border": "2px solid black",
    },
  },
}
`;

exports[`unify with utilities 1`] = `
".test {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

@media (min-width: 640px) {
  .test {
    border-width: 1px;
    border-radius: 0.25rem;
    background-color: rgba(96, 165, 250, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
}"
`;

exports[`unify with utilities 2`] = `
".test2 {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  font-size: 1.125rem;
  line-height: 1.75rem;
  border-radius: 0.375rem;
  border-width: 2px;
}

@media (min-width: 640px) {
  .test2 {
    border-width: 1px;
    border-radius: 0.25rem;
    background-color: rgba(96, 165, 250, var(--w-bg-opacity));
    --w-bg-opacity: 1;
  }
}"
`;

exports[`unify with utility object 1`] = `
".test {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.button {
  background-color: rgba(244, 63, 94, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  border-radius: 0.25rem;
  border-width: 1px;
}"
`;

exports[`unify with utility object: a 1`] = `
".test {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.button {
  background-color: rgba(244, 63, 94, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  border-radius: 0.25rem;
  border-width: 1px;
}

p {
  font-size: 1.125rem;
  line-height: 1.75rem;
}"
`;

exports[`unify with utility object: a 2`] = `
".test {
  background-color: rgba(59, 130, 246, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.button {
  background-color: rgba(244, 63, 94, var(--w-bg-opacity));
  --w-bg-opacity: 1;
  border-radius: 0.25rem;
  border-width: 1px;
}

p {
  font-size: 1.125rem;
  line-height: 1.75rem;
}"
`;
